query-engine/query-engine-node-api/src/engine.rs

```diff
@@ -416,17 +413,25 @@ impl QueryEngine {
     /// Loads the query schema. Only available when connected.
     #[napi]
     pub async fn sdl_schema(&self) -> napi::Result<String> {
+        let dispatcher = self.logger.dispatcher();
+        let recorder = self.logger.recorder();
+
         async_panic_to_js_error(async move {
             let inner = self.inner.read().await;
             let engine = inner.as_engine()?;
 
             Ok(render_graphql_schema(engine.query_schema()))
         })
+        .with_subscriber(dispatcher)
+        .with_optional_recorder(recorder)
         .await
     }
 
     #[napi]
     pub async fn metrics(&self, json_options: String) -> napi::Result<String> {
+        let dispatcher = self.logger.dispatcher();
+        let recorder = self.logger.recorder();
```

Comment:
We've updated the `query-engine-node-api` and `query-engine` crates. What about the `query-engine-c-abi` one, is there anything to refactor in there in terms of metrics setup?

Reply:
`query-engine-c-abi` and `query-engine-wasm` don't support metrics and have no metrics setup code.